pkg/orchestrator/gitHubActions.go

```diff
@@ -1,88 +1,227 @@
 package orchestrator
 
 import (
-	"os"
+	"bytes"
+	"context"
+	"fmt"
+	piperHttp "github.com/SAP/jenkins-library/pkg/http"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sync/semaphore"
+	"io"
 	"strings"
+	"sync"
 	"time"
 
 	"github.com/SAP/jenkins-library/pkg/log"
 )
 
-type GitHubActionsConfigProvider struct{}
+type GitHubActionsConfigProvider struct {
+	client piperHttp.Client
+	run    run
+}
+
+func getActionsURL() string {
+	ghURL := getEnv("GITHUB_URL", "")
+	switch ghURL {
+	case "https://github.com/":
+		ghURL = "https://api.github.com"
+	default:
+		ghURL += "api/v3"
+	}
+	return fmt.Sprintf("%s/repos/%s/actions", ghURL, getEnv("GITHUB_REPOSITORY", ""))
+}
+
+func initGHProvider(settings *OrchestratorSettings) (*GitHubActionsConfigProvider, error) {
+	g := GitHubActionsConfigProvider{}
+	g.client = piperHttp.Client{}
+	g.client.SetOptions(piperHttp.ClientOptions{
+		Password:         settings.GitHubToken,
+		MaxRetries:       3,
+		TransportTimeout: time.Second * 10,
+	})
+	return &g, nil
+}
+
+func (g *GitHubActionsConfigProvider) getData() error {
+	resp, err := g.client.GetRequest(
+		fmt.Sprintf(
+			"%s/runs/%s", getActionsURL(), getEnv("GITHUB_RUN_ID", ""),
+		),
+		map[string][]string{
+			"Accept":        {"application/vnd.github+json"},
+			"Authorization": {"Bearer $GITHUB_TOKEN"},
+		}, nil)
+	if err != nil || resp.StatusCode != 200 {
+		return fmt.Errorf("can't get API data: %w", err)
+
+	}
+	err = piperHttp.ParseHTTPResponseBodyJSON(resp, &g.run)
+	if err != nil {
+		return fmt.Errorf("can't parse JSON data: %w", err)
+	}
 
-func (g *GitHubActionsConfigProvider) InitOrchestratorProvider(settings *OrchestratorSettings) {
-	log.Entry().Debug("Successfully initialized GitHubActions config provider")
+	return nil
 }
 
 func (g *GitHubActionsConfigProvider) OrchestratorVersion() string {
+	log.Entry().Debugf("OrchestratorVersion() for GitHub Actions is not applicable.")
 	return "n/a"
 }
 
 func (g *GitHubActionsConfigProvider) OrchestratorType() string {
-	return "GitHubActions"
+	return "GitHub Actions"
 }
 
 func (g *GitHubActionsConfigProvider) GetBuildStatus() string {
-	log.Entry().Infof("GetBuildStatus() for GitHub Actions not yet implemented.")
-	return "FAILURE"
+	// By default, we will assume it's a success
```

Comment:
I would not assume that this will be handled by actions itself. The GetBuildStatus method will be also used e.g. inside the reporting step in GPP to get the overall build status of a pipeline. Not sure if this is covered by Actions then?